codecs/qoi/enc/qoi_enc.cpp

```diff
@@ -0,0 +1,34 @@
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+#define QOI_IMPLEMENTATION
+#include "qoi.h"
+
+using namespace emscripten;
+
+struct QoiOptions {};
+
+thread_local const val Uint8Array = val::global("Uint8Array");
+
+val encode(std::string buffer, int width, int height, QoiOptions options) {
+  int compressedSizeInBytes;
+  qoi_desc desc;
+  desc.width = width;
+  desc.height = height;
+  desc.channels = 4;
+  desc.colorspace = QOI_SRGB;
+
+  void* encodedData = qoi_encode(buffer.c_str(), &desc, &compressedSizeInBytes);
+  if (encodedData == NULL)
+    return val::null();
+
+  auto js_result =
+      Uint8Array.new_(typed_memory_view(compressedSizeInBytes, (const uint8_t*)encodedData));
```

Comment:
This is leaking memory. From the `qoi.h::qoi_encode()` docs:

> The returned qoi data should be free()d after use.

It’s not a big deal as we are killing the Wasm modules quite regularly, but if someone is encoding really big images in fast succession, it could lead to an OOM error.

```suggestion
      Uint8Array.new_(typed_memory_view(compressedSizeInBytes, (const uint8_t*)encodedData));
      free(encodedData);
```

Reply:
Ohh, nice catch! Committing your suggestion directly 👍